Ark: Survival Ascended

dagbs opened the pull request:
# Description

Add support for Ark: Survival Ascended egg

fixes: https://github.com/parkervcp/eggs/issues/2523

## Checklist for all submissions

<!-- insert X into the brackets to mark it as done (i.e. [x]). You can click preview to make the links appear clickable. -->

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

<!-- You can erase the new egg submission template if you're not adding a completely new egg -->

## New egg Submissions

1. [x] Does your submission pass tests (server is connectable)?
2. [ ] Does your egg use a custom docker image?
    * [ ] Have you tried to use a generic image?
    * [ ] Did you PR the necessary changes to make it work?
3. [x] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [x] Have you added a unique README.md for the egg you are adding according to the alphabetical order?
5. [x] You verify that the start command applied does not use a shell script
    * [ ] If some script is needed then it is part of a current yolk or a PR to add one
6. [x] The egg was exported from the panel

dagbs commented:
https://github.com/parkervcp/eggs/issues/2523

gOOvER commented:
ty :)

Log1x commented on game_eggs/steamcmd_servers/ark_survival_ascended/egg-ark--survival-ascended.json:
```diff
@@ -0,0 +1,132 @@
+        },
+        {
+            "name": "Additional Arguments (FLAGS)",
+            "description": "params (?ServerPassword=...) and flags (-UseBattleEye) are supported here but params need to be first and touching followed by flags without them touching. (i.e): ?ServerPassword=MyP4SsWoRd?ServerAdminPassword=sUpRp4ssW0rd -UseBattleEye -cossplay",
```
This description could probably be rewritten. Params should generally never be here since they'd be outside of the quotes in the start command.

dagbs replied:
I wanted to leave the description for both and label them so people knew the difference between PARAMS & FLAGS. params stay inside the quotes and need to be touching where as flags are outside the quotes and do not touch

Log1x replied:
Yeah but right now it implies that both are supported which they wouldn't be since with the current start command wrapping the parameters in quotes, there's no way for you to pass parameters to Flags like the description says. They have to be the first argument passed to the binary.

Log1x commented on game_eggs/steamcmd_servers/ark_survival_ascended/egg-ark--survival-ascended.json:
```diff
@@ -0,0 +1,132 @@
+    "startup": "proton run .\/ShooterGame\/Binaries\/Win64\/ArkAscendedServer.exe \"{{SERVER_MAP}}?listen?MaxPlayers={{MAX_PLAYERS}}?SessionName={{SESSION_NAME}}?Port={{SERVER_PORT}}?QueryPort={{QUERY_PORT}}?ServerAdminPassword={{ARK_ADMIN_PASSWORD}}{{ARGS_PARAMS}}\"$( [ \"$BATTLE_EYE\" == \"1\" ] || printf %s ' -NoBattlEye' ) {{ARGS_FLAG}} & ARK_PID=$! ; tail -F .\/ShooterGame\/Saved\/Logs\/Shoo ...
+    "config": {
+        "files": "{}",
+        "startup": "{\r\n    \"done\": \"tail: .\/ShooterGame\/Saved\/Logs\/ShooterGame.log: file truncated\"\r\n}",
```
Can we check for the server starting instead?

Maybe `" 0]Full Startup: "`

dagbs replied:
im working on doing a sleep timer which looks to be working. so, I'll do a commit shortly to add it (just seeing how long i need to sleep for it to work

gOOvER commented on game_eggs/steamcmd_servers/ark_survival_ascended/egg-ark--survival-ascended.json:
```diff
@@ -0,0 +1,132 @@
+    "name": "Ark: Survival Ascended",
+    "author": "blood@darkartsgaming.com",
+    "description": "ARK is reimagined from the ground-up into the next-generation of video game technology with Unreal Engine 5! Form a tribe, tame & breed hundreds of unique dinosaurs and primeval creatures, explore, craft, build, and fight your way to the top of the food-chain. Your new world awaits!",
+    "features": null,
```
i would add here:

```
"features": [
        "steam_disk_space"
    ],
```

its only possible, when you edit egg oputside ptero

dagbs replied:
done. i was wondering how to do that from inside the panel

gOOvER replied:
> done. i was wondering how to do that from inside the panel

not possible yet. But maybe sometimes :D

Log1x commented on game_eggs/steamcmd_servers/ark_survival_ascended/egg-ark--survival-ascended.json:
```diff
@@ -0,0 +1,132 @@
+        "ghcr.io\/parkervcp\/steamcmd:proton": "ghcr.io\/parkervcp\/steamcmd:proton"
+    },
+    "file_denylist": [],
+    "startup": "proton run .\/ShooterGame\/Binaries\/Win64\/ArkAscendedServer.exe \"{{SERVER_MAP}}?listen?MaxPlayers={{MAX_PLAYERS}}?SessionName={{SESSION_NAME}}?Port={{SERVER_PORT}}?QueryPort={{QUERY_PORT}}?ServerAdminPassword={{ARK_ADMIN_PASSWORD}}{{ARGS_PARAMS}}\"$( [ \"$BATTLE_EYE\" == \"1\" ] || printf %s ' -NoBattlEye' ) {{ARGS_FLAG}} & ARK_PID=$! ; tail -F .\/ShooterGame\/Saved\/Logs\/Shoo ...
```
Probably change to:

```sh
proton run .\/ShooterGame\/Binaries\/Win64\/ArkAscendedServer.exe \"{{SERVER_MAP}}?listen?MaxPlayers={{MAX_PLAYERS}}?SessionName={{SESSION_NAME}}?Port={{SERVER_PORT}}?QueryPort={{QUERY_PORT}}?ServerAdminPassword={{ARK_ADMIN_PASSWORD}}{{ARGS_PARAMS}}\" $( [ \"$BATTLE_EYE\" == \"1\" ] || printf %s ' -NoBattlEye ' ){{ARGS_FLAG}} & ARK_PID=$! ; tail -F .\/ShooterGame\/Saved\/Logs\/ShooterGame.log
```

Adding a space before the battle eye conditional, appending a space to `-NoBattlEye ` and then removing the space between `ARGS_FLAGS` to always maintain consistency.

dagbs replied:
there needs to be a space after {{ARGS_FLAG}} and not before it if battleeye is enabled & args are not set to prevent multiple spaces before the `& ARK_PID...`

Log1x replied:
I think this is fine now with your flags fix. My review just came in late.

Log1x commented on game_eggs/steamcmd_servers/ark_survival_ascended/egg-ark--survival-ascended.json:
```diff
@@ -0,0 +1,134 @@
+        },
+        {
+            "name": "Additional Arguments (PARAMS)",
+            "description": "params (?ServerPassword=...) and flags (-UseBattleEye) are supported here but params need to be first and touching followed by flags without them touching. (i.e): ?ServerPassword=MyP4SsWoRd?ServerAdminPassword=sUpRp4ssW0rd -UseBattleEye -cossplay",
```
Probably needs rewritten since this should typically only be arguments.

Log1x commented on game_eggs/steamcmd_servers/ark_survival_ascended/README.md:
```diff
@@ -0,0 +1,23 @@
+# Ark: Survival Ascended
```
ARK should be capitalized.

Ballaual replied:
arkse is writting same way here - would keep it consistent?

Log1x replied:
I hadn't noticed. I'd just fix it there too considering it's the branded stylization of the game's name but otherwise whatever.

Log1x commented on game_eggs/steamcmd_servers/ark_survival_ascended/egg-ark--survival-ascended.json:
```diff
@@ -0,0 +1,134 @@
+        "update_url": null
+    },
+    "exported_at": "2023-10-28T14:43:36-04:00",
+    "name": "Ark: Survival Ascended",
```
ARK should be capitalized.

Ballaual replied:
arkse is writting same way here - would keep it consistent?

Log1x commented:
Only request outside of my review would be a Mods environment variable for `-mods=` -- otherwise everything is working great. Thanks 😄

Ballaual commented:
not absolutely necessary but i would adjust the startconfig like this. so you have 100% feedback that the server is really running?

```
{
    "done": "has successfully started!"
}
```

Log1x commented:
> not absolutely necessary but i would adjust the startconfig like this. so you have 100% feedback that the server is really running?

Mentioned it in my review above and added what should be a solid alternative.

dagbs commented:
> not absolutely necessary but i would adjust the startconfig like this. so you have 100% feedback that the server is really running?
> 
> ```
> {
>     "done": "has successfully started!"
> }
> ```

The problem with using `tail -F ...` is it pulls the file as it was. It doesn't get reset until after the server is started and the text of "has successfully started!" is in the previous file which would provide a more false-start than it already is. I don't have a good way of cleaning the previous log file without loosing data or adding even more cleanup/organization code.

hackles commented on game_eggs/steamcmd_servers/ark_survival_ascended/egg-ark--survival-ascended.json:
```diff
@@ -0,0 +1,134 @@
+    "startup": "proton run .\/ShooterGame\/Binaries\/Win64\/ArkAscendedServer.exe \"{{SERVER_MAP}}?listen?MaxPlayers={{MAX_PLAYERS}}?SessionName={{SESSION_NAME}}?Port={{SERVER_PORT}}?QueryPort={{QUERY_PORT}}?ServerAdminPassword={{ARK_ADMIN_PASSWORD}}{{ARGS_PARAMS}}\"$( [ \"$BATTLE_EYE\" == \"1\" ] || printf %s ' -NoBattlEye' ) {{ARGS_FLAG}} & ARK_PID=$! ; tail -F .\/ShooterGame\/Saved\/Logs\/Shoo ...
+    "config": {
+        "files": "{}",
+        "startup": "{\r\n    \"done\": \"tail: .\/ShooterGame\/Saved\/Logs\/ShooterGame.log: file truncated\"\r\n}",
```
```suggestion
        "startup": "{\r\n    \"done\": \"\\\" has successfully started!\"\r\n}",
```
This checks for when the server has started and is ready for players to join.

Log1x commented:
> The problem with using tail -F ... is it pulls the file as it was.

Good point. Could always make it sleep for a second so the server has time to rename the log? I assume it happens very quickly but I'm not sure.

Log1x commented:
GitHub apparently has me talking to myself...

![Screenshot](https://i.imgur.com/LKSnuYZ.png)
![Screenshot](https://i.imgur.com/8RYcfrZ.png)

Ballaual commented:
> > The problem with using tail -F ... is it pulls the file as it was.
> 
> Good point. Could always make it sleep for a second so the server has time to rename the log? I assume it happens very quickly but I'm not sure.

would it delete and recreate instead? add 1 sec sleep before and then execute?



> GitHub apparently has me talking to myself...
already thought which reviews xD

gOOvER commented:
> Edit: Nevermind I'm dumb. :)

i know that :D when i forget to press a button xD

Ballaual reviewed (commented):
missing entry in following readme files

https://github.com/parkervcp/eggs/tree/master/game_eggs
https://github.com/parkervcp/eggs/tree/master/game_eggs/steamcmd_servers

hackles commented:
> > The problem with using tail -F ... is it pulls the file as it was.
> 
> Good point. Could always make it sleep for a second so the server has time to rename the log? I assume it happens very quickly but I'm not sure.

Can confirm that this works. 👍

`... ; sleep 5 && tail ...`

Thanks @Log1x

hackles commented on game_eggs/steamcmd_servers/ark_survival_ascended/egg-ark--survival-ascended.json:
```diff
@@ -0,0 +1,134 @@
+        "ghcr.io\/parkervcp\/steamcmd:proton": "ghcr.io\/parkervcp\/steamcmd:proton"
+    },
+    "file_denylist": [],
+    "startup": "proton run .\/ShooterGame\/Binaries\/Win64\/ArkAscendedServer.exe \"{{SERVER_MAP}}?listen?MaxPlayers={{MAX_PLAYERS}}?SessionName={{SESSION_NAME}}?Port={{SERVER_PORT}}?QueryPort={{QUERY_PORT}}?ServerAdminPassword={{ARK_ADMIN_PASSWORD}}{{ARGS_PARAMS}}\"$( [ \"$BATTLE_EYE\" == \"1\" ] || printf %s ' -NoBattlEye' ) {{ARGS_FLAGS}} & ARK_PID=$! ; tail -F .\/ShooterGame\/Saved\/Logs\/Sho ...
```
```suggestion
    "startup": "proton run .\/ShooterGame\/Binaries\/Win64\/ArkAscendedServer.exe \"{{SERVER_MAP}}?listen?MaxPlayers={{MAX_PLAYERS}}?SessionName={{SESSION_NAME}}?Port={{SERVER_PORT}}?QueryPort={{QUERY_PORT}}?ServerAdminPassword={{ARK_ADMIN_PASSWORD}}{{ARGS_PARAMS}}\"$( [ \"$BATTLE_EYE\" == \"1\" ] || printf %s ' -NoBattlEye' ) {{ARGS_FLAGS}} & ARK_PID=$! ; sleep 5 && tail -F .\/ShooterGame\/Saved\/Logs\/ShooterGame.log",
```

This resolves issue with `tail` picking up old log file (h/t @Log1x)

Ballaual replied:
```suggestion
    "startup": "proton run .\/ShooterGame\/Binaries\/Win64\/ArkAscendedServer.exe \"{{SERVER_MAP}}?listen?MaxPlayers={{MAX_PLAYERS}}?SessionName={{SESSION_NAME}}?ServerPassword={{SERVER_PASSWORD}}?Port={{SERVER_PORT}}?QueryPort={{QUERY_PORT}}?ServerAdminPassword={{ARK_ADMIN_PASSWORD}}{{ARGS_PARAMS}}\"$( [ \"$BATTLE_EYE\" == \"1\" ] || printf %s ' -NoBattlEye' ) {{ARGS_FLAGS}} & ARK_PID=$! ; sleep 5 && tail -F .\/ShooterGame\/Saved\/Logs\/ShooterGame.log",
```
adding serverpassword to startup command

vp-en commented:
Sorry if this isn't relevant, or if my code is shitty, but couldn't you also add a separate password variable and then modify the startup command like so:

**Variable:**
```

```

****

Ballaual commented:
> Sorry if this isn't relevant, or if my code is shitty, but couldn't you also add a separate password variable and then modify the startup command like so:
> 
> **Variable:**
> 
> ```
> {
>     "name": "Server Password",
>     "description": "The password for players to join the server.",
>     "env_variable": "SERVER_PASSWORD",
>     "default_value": "",
>     "user_viewable": true,
>     "user_editable": true,
>     "rules": "nullable|alpha_dash|max:32",
>     "field_type": "text"
> }
> ```
> 
> **Startup Command:**
> 
> ```
> "startup": "proton run .\/ShooterGame\/Binaries\/Win64\/ArkAscendedServer.exe \"{{SERVER_MAP}}?listen?MaxPlayers={{MAX_PLAYERS}}?SessionName={{SESSION_NAME}}?Port={{SERVER_PORT}}?QueryPort={{QUERY_PORT}}?ServerAdminPassword={{ARK_ADMIN_PASSWORD}}$( [ \"${SERVER_PASSWORD}\" != \"\" ] && printf %s \"?ServerPassword=${SERVER_PASSWORD}\" ){{ARGS_PARAMS}}\"$( [ \"$BATTLE_EYE\" == \"1\" ] || printf %s ' -NoBattlEye' ) {{ARGS_FLAGS}} & ARK_PID=$! ; tail -F .\/ShooterGame\/Saved\/Logs\/ShooterGame.log",
> ```

true thats missing..
but start command does not need to be that complicated just add ?ServerPassword={{SERVER_PASSWORD}}

vp-en commented:
> @dagbs true thats missing.. but start command does not need to be that complicated just add ?ServerPassword={{SERVER_PASSWORD}}

Oh. I made it like that, since I was wondering whether or not the server software would accept an empty string for password, or not

Ballaual commented on game_eggs/steamcmd_servers/ark_survival_ascended/egg-ark--survival-ascended.json:
```diff
@@ -0,0 +1,134 @@
+            "user_editable": true,
+            "rules": "nullable|string",
+            "field_type": "text"
+        }
```
```suggestion
        },
        {
            "name": "Server Password",
            "description": "If specified, players must provide this password to join the server.",
            "env_variable": "SERVER_PASSWORD",
            "default_value": "",
            "user_viewable": true,
            "user_editable": true,
            "rules": "nullable|alpha_dash|between:1,100",
            "field_type": "text"
        }
```
adds server password variable

Ballaual reviewed (changes requested):
this adds support for server password

dagbs commented:
Latest commit addresses all above mentioned issues.

If there is something missing, it's because it didn't work as expected and/or can be done using existing PARAMS/FLAGS option.

Ballaual commented:
@dagbs just found a small typo inside both additional args :) 
at the end you write -cossplay instead of -crossplay :p
aaaand just a little understanding question for pve/pvp variable
-> name is server pve and enabled per default right?
in description you write default is pvp - either description or default value seems a bit misleading ^^

IxPrumxI commented:
I just wanted to ask. The stop seems to be force quitting the process are we sure that it actually saves the game when doing that or do we risk losing player progress? Because looking at ARK:SE it uses trap and rcon to save the wolrd and quit the game afterwards.

Ballaual commented:
> I just wanted to ask. The stop seems to be force quitting the process are we sure that it actually saves the game when doing that or do we risk losing player progress? Because looking at ARK:SE it uses trap and rcon to save the wolrd and quit the game afterwards.

that question i am asking myself for the whole time ^^ unfortunately i am not able to test it myself.
tested a lot with rcon and seems not working for now :/

dagbs commented:
> additional args: small typo inside both variables :) at the end you write -cossplay instead of -crossplay :p

fixed in https://github.com/parkervcp/eggs/pull/2529/commits/4466587549fb4c0fbd66aacc9a759d938872bb9b thanks

> pve/pvp: -> name is server pve and enabled per default right? in description you write default is pvp - either description or default value seems a bit misleading ^^ and to keep is consistent either write both sentences with "x-vs-x" or "x vs x" ;)

fixed in https://github.com/parkervcp/eggs/pull/2529/commits/4466587549fb4c0fbd66aacc9a759d938872bb9b thanks

> auto update: This is to enable auto-updating for servers. Default is 1. Set to 1 to update -> either remove "Set to 1 to update" or do sth like this: "Set to 0 to disable automatic updates." and the appId should not be editable by the user itself, can be read only or completely hidden

i forgot to update the description properly and just made quick changes, updated in https://github.com/parkervcp/eggs/pull/2529/commits/4466587549fb4c0fbd66aacc9a759d938872bb9b

> mods: any chance to add modsupport?

unfortunalty, as I mentioned before, it's not "officially" documented and my tests yielded mixed results. the option to _add_ mods is still available through Additional Args (PARAMS/FLAGS) which is why that's available. I also added troubleshooting in the readme because of this since there was no _clear_ reason why it did or didn't work and if you mess it up it's difficult to identify what went wrong.

> I just wanted to ask. The stop seems to be force quitting the process are we sure that it actually saves the game when doing that or do we risk losing player progress? Because looking at ARK:SE it uses trap and rcon to save the wolrd and quit the game afterwards.

that's a good question and the answer is: rcon isn't officially supported. I tried many times to get it working but that's the best I can do as of now, I'm sorry. if you want to setup a schedule with autobackups, you can make a file called `/home/container/.pteroignore` and just put:
```
*
!ShooterGame/Saved/*
```
this way you can keep backups. just know that there is a timer in ark of 15 minutes by default so doing a ptero backup more often than this without changing this setting wont do anything.

Ballaual commented:
maybe the adjustment of the appid variable (remove user write access) then i am completely happy :D
further changes such as rcon and mods we can update later if its fully implemented and working

Ballaual commented:
❤️ from my perspective nothing more to complain right now!✅

Log1x commented:
> is it running same way arkse does? did not tried it out yet..
> ?GameModIds={{MOD_ID}} and -automanagedmods or really just -mods={{MOD_ID}}

Just `-mods={{MOD_IDS}}` 

Currently running `-mods=900062,908148,912902,914844,916922,926956,927131` without any issues with multiple players connected at the moment.

vp-en commented:
Anyone having trouble connecting to their server? I can't seem to connect to mine 😭 . And even though I've set max_players to 20, the server browser still reports 70, which I guess is the default.

Ballaual commented:
> Yes the only way to truly prevent losing progress would be to do `AdminCheat saveworld` in-game before stopping at the moment.

this should be added to the servers README.md then? Just in case no one can say its nowhere marked ^^

Log1x commented on game_eggs/steamcmd_servers/ark_survival_ascended/egg-ark--survival-ascended.json:
```diff
@@ -0,0 +1,154 @@
+        "ghcr.io\/parkervcp\/steamcmd:proton": "ghcr.io\/parkervcp\/steamcmd:proton"
+    },
+    "file_denylist": [],
+    "startup": "proton run .\/ShooterGame\/Binaries\/Win64\/ArkAscendedServer.exe \"{{SERVER_MAP}}?listen?MaxPlayers={{MAX_PLAYERS}}?SessionName={{SESSION_NAME}}?Port={{SERVER_PORT}}?QueryPort={{QUERY_PORT}}?ServerAdminPassword={{ARK_ADMIN_PASSWORD}}$( [  \"$SERVER_PVE\" == \"0\" ] || printf %s '?ServerPVE=True' )$( [ -z \"$SERVER_PASSWORD\" ] || printf %s '?ServerPassword={{SERVER_PASSWORD}}' ){ ...
```
We can actually pull this off without `sleep` using only `tail` by passing `-c0` to not show previous bytes.

```
tail -c0 -F ./ShooterGame/Saved/Logs/ShooterGame.log --pid=$ARK_PID
```

We can also pass the `$ARK_PID` to `--pid=` to track if the server crashes/exits as it will kill `tail` accordingly.

🎉

dagbs replied:
Interesting, I figured something like this was possible.
I'll test and verify it works and will do a commit for this.

dagbs commented:
> > Yes the only way to truly prevent losing progress would be to do `AdminCheat saveworld` in-game before stopping at the moment.
> 
> this should be added to the servers README.md then? Just in case no one can say its nowhere marked ^^

`saveworld` is not documented officially by Ark so I will not be including it in any README

Log1x commented:
> Anyone having trouble connecting to their server? I can't seem to connect to mine 😭 . And even though I've set max_players to 20, the server browser still reports 70, which I guess is the default.

No issues connecting on my end. But I did notice MaxPlayers forcing 70 even when specifying it in the start command as well as the config.

vp-en commented:
> > Anyone having trouble connecting to their server? I can't seem to connect to mine 😭 . And even though I've set max_players to 20, the server browser still reports 70, which I guess is the default.
> 
> No issues connecting on my end. But I did notice MaxPlayers forcing 70 even when specifying it in the start command/config.

Seems to work now, but there might be something up with the parsing in the startup command. 
I've set a password, and it's saying Invalid Password when connecting. I'm omw to bed, so I can't really test more for now though 😄

dagbs commented:
> > > Anyone having trouble connecting to their server? I can't seem to connect to mine 😭 . And even though I've set max_players to 20, the server browser still reports 70, which I guess is the default.
> > 
> > 
> > No issues connecting on my end. But I did notice MaxPlayers forcing 70 even when specifying it in the start command/config.
> 
> Seems to work now connecting-wise 🙂 I've set a password, and it's saying Invalid Password when connecting. That, combined with the max_players not seeming to apply, leads me to believe something might be up with the startup command.
> 
> I'm omw to bed, so I can't really test more for now though 😄

It might be an issue with how it re-saves params in the `GameUserSettings.ini` if you've started the server once already.

vp-en commented:
> > > > Anyone having trouble connecting to their server? I can't seem to connect to mine 😭 . And even though I've set max_players to 20, the server browser still reports 70, which I guess is the default.
> > > 
> > > 
> > > No issues connecting on my end. But I did notice MaxPlayers forcing 70 even when specifying it in the start command/config.
> > 
> > 
> > Seems to work now connecting-wise 🙂 I've set a password, and it's saying Invalid Password when connecting. That, combined with the max_players not seeming to apply, leads me to believe something might be up with the startup command.
> > I'm omw to bed, so I can't really test more for now though 😄
> 
> It might be an issue with how it re-saves params in the `GameUserSettings.ini` if you've started the server once already.

This is from a fresh-server install from scratch, from the latest commit as of this writing. Sorry if forgot to include that in the comment :).

KJairen commented:
regarding the ASA egg, I had it working perfectly fine earlier today, but for the past 2 - 3 hours I've just been stuck trying everything I can to connect to my server.

I have tried reinstalling, deleting all my files + reinstalling, changing ports, changing back my ports (both times the correct ports were correctly port-forwarded and assigned) and I also just deleted my server and did a fresh install with this newer egg.

No matter what I do, I just cannot manage to join my server anymore, even though I didn't change anything as far as I'm aware.
the server shows in the unofficial server list, but when I join it, it just says "Connection Timeout", while I am able to join other servers.

I thought this might've been due to the mods saying they're "outdated" but joining any other servers, even with "outdated" mods works perfectly fine

Any ideas?

Ballaual commented:
> he maxplayers is unchanged `70`, even though I've set it to 20 in the startup options. And the server password is just `ServerPassword="${SERVER_PASSWORD}"` - so the code I provided might be wrong. Sorry about that.

just for testing could u pls try out with that start command?
`proton run ./ShooterGame/Binaries/Win64/ArkAscendedServer.exe {{SERVER_MAP}}?listen?MaxPlayers={{MAX_PLAYERS}}?SessionName="{{SESSION_NAME}}"?Port={{SERVER_PORT}}?QueryPort={{QUERY_PORT}}?ServerAdminPassword={{ARK_ADMIN_PASSWORD}}$( [  "$SERVER_PVE" == "0" ] || printf %s '?ServerPVE=True' ) {{ARGS_PARAMS}} $( [ "$BATTLE_EYE" == "1" ] || printf %s ' -NoBattlEye' ) {{ARGS_FLAGS}} & ARK_PID=$! ; sleep 15 ; tail -F ./ShooterGame/Saved/Logs/ShooterGame.log`

Log1x commented:
Looks like `-servergamelog` is pretty necessary if you want any kind of useful log output during game sessions.

`-servergamelogincludetribelogs` and `-NotifyAdminCommandsInChat` are also useful. The latter will give you (and other admins) feedback in-game when you execute a command.

vp-en commented:
> server password was missing here .. sorry... here the updated line for testing `proton run ./ShooterGame/Binaries/Win64/ArkAscendedServer.exe {{SERVER_MAP}}?listen?MaxPlayers={{MAX_PLAYERS}}?SessionName="{{SESSION_NAME}}"?Port={{SERVER_PORT}}?QueryPort={{QUERY_PORT}}?ServerPassword={{SERVER_PASSWORD}}?ServerAdminPassword={{ARK_ADMIN_PASSWORD}}$( [ "$SERVER_PVE" == "0" ] || printf %s '?ServerPVE=True' ) {{ARGS_PARAMS}} $( [ "$BATTLE_EYE" == "1" ] || printf %s ' -NoBattlEye' ) {{ARGS_FLAGS}} & ARK_PID=$! ; sleep 15 ; tail -F ./ShooterGame/Saved/Logs/ShooterGame.log`

Server password now applies, but the maxplayers still isn't. Not sure if it's an "egg" (startup command) issue or the the dedicated server software issue itself. Now I'm for real on my way to bed - but thanks for fixing it so quickly :)

Ballaual commented:
i removed the maxplayers in startup and changed the value inside the config by hand.
after restart it changed back to 70

[/Script/Engine.GameSession]
MaxPlayers=20

even my nitrado server with hardcoded 20 in the config file shows up with 0/70

IxPrumxI commented:
Hmmm it likes to grab more than it needs lol
![image](https://github.com/parkervcp/eggs/assets/34715358/b407b960-b1a9-4c32-8e43-eb871de00ca8)

```
[2023.10.29-00.01.02:804][  0]Server: "PrumLinux" has successfully started!
[2023.10.29-00.01.05:148][  0]Commandline:  TheIsland_WP?listen?MaxPlayers=70?SessionName=PrumLinux?Port=7780?QueryPort=27069?ServerAdminPassword=changeme?ServerPVE=True?ServerPassword=123 -servergamelog -servergamelogincludetribelogs -NotifyAdminCommandsInChat
[2023.10.29-00.01.05:148][  0]Full Startup: 68.66 seconds
```

dagbs commented:
> The maxplayers is unchanged `70`, even though I've set it to 20 in the startup options. And the server password is just `ServerPassword="${SERVER_PASSWORD}"` - so the code I provided might be wrong. Sorry about that.

As for the ServerPassword, I just verified this was an issue, this is fixed in: https://github.com/parkervcp/eggs/pull/2529/commits/80ba8baaf91cec0fc01d138b349e48427d2e543f

As for MaxPlayers, there is nothing more we can do to fix this issue. Edit your `GameUserSettings.ini` and start.

-----

To those that are having issues, please make sure your startup command is using the latest one. After creating a server, updating the egg doesn't change existing server's startup commands; they must be updated manually.

> was able to fix it using (. . .)

This was modified in the latest commit along with fixing the `tail -F...` command, thanks to @Log1x

Ballaual commented:
> As for MaxPlayers, there is nothing more we can do to fix this issue. As far as I know, this is not an issue with the egg directly.
the server seems to ignore the MaxPlayers or overiding it every restart.

only attempt we maybe can try is to pass the MAX_PLAYERS within Configuration Files section

dont know if its possible for ini files because i didnt get managed it so far. or did not found the correct way to parse them..
here an example of how subnautica is doing it
```
{
    "server.cfg": {
        "parser": "properties",
        "find": {
            "ServerPort": "{{server.build.default.port}}"
        }
    }
}
```
here it should be sth like:
```
{
    "ShooterGame/Saved/Config/GameUserSettings.ini": {
        "parser": "properties",
        "find": {
            "MaxPlayers": "{{server.build.env.MAX_PLAYERS}}"
        }
    }
}
```
but this is wiping the whole file and just returns the MaxPlayers=value

Log1x commented:
I think you can just use `ini` for the `parser`.

Ballaual commented:
i missed that the correct path is "ShooterGame/Saved/Config/WindowsServer/GameUserSettings.ini" btw

Ballaual commented:
> ```json
> {
>     "ShooterGame/Saved/Config/WindowsServer/GameUserSettings.ini": {
>         "parser": "ini",
>         "find": {
>             "/Script/Engine.GameSession.MaxPlayers": "{{server.build.env.MAX_PLAYERS}}"
>         }
>     }
> }
> ```
returns this and messed up file as in the screenshot above
```
[/Script/Engine.GameSession]
MaxPlayers = 70

[/Script/Engine]
GameSession.MaxPlayers = 12
```

Log1x commented:
> using ini seems to fuck this even more up xD

Ahh fun. Could maybe use `file` then for the `parser`. I've never used it before though. I can help with testing this stuff later but in the mean time my wife started building her house so the server is off limits. 😆 

```
{
    "ShooterGame/Saved/Config/WindowsServer/GameUserSettings.ini": {
        "parser": "file",
        "find": {
            "MaxPlayers=": "MaxPlayers={{server.build.env.MAX_PLAYERS}}"
        }
    }
}
```

Log1x commented:
For `ini` you could also try:

```
"/Script/Engine\.GameSession.MaxPlayers": "{{server.build.env.MAX_PLAYERS}}"
```

Ballaual commented:
will test / try out tomorow, gotta go sleep for now ^^

Ballaual commented:
> ```json
> "/Script/Engine\.GameSession.MaxPlayers": "{{server.build.env.MAX_PLAYERS}}"
> ```

this can't be saved, panel says "The config files must be a valid JSON string."

dagbs commented:
> Ahh fun. Could maybe use `file` then for the `parser`. I've never used it before though. I can help with testing this stuff later but in the mean time my wife started building her house so the server is off limits. 😆
> 
> ```json
> {
>     "ShooterGame/Saved/Config/WindowsServer/GameUserSettings.ini": {
>         "parser": "file",
>         "find": {
>             "MaxPlayers=": "MaxPlayers={{server.build.env.MAX_PLAYERS}}"
>         }
>     }
> }
> ```

This looks to be the best option in combination with the startup command. I tried to remove the param and it failed on a fresh install; but, both together, it works.

Ballaual commented:
@dagbs could you pls check if you get the same problem issue with latest version 
look ServerAdminPassword

![image](https://github.com/parkervcp/eggs/assets/38478976/8625063b-449a-4d03-a63b-f56be59e0410)

dagbs commented:
> @dagbs could you pls check if you get the same problem issue with latest version look ServerAdminPassword

Are you using the latest startup command for your server?

Ballaual commented:
yes, on startup all fine, but when marked as running it messes up

dagbs commented:
> yes, on startup all fine, but when marked as running it messes up

in your `GameUserSettings.ini` delete the `ServerAdminPassword=` line and let it regenerate.

Ballaual commented:
> > yes, on startup all fine, but when marked as running it messes up
> 
> in your `GameUserSettings.ini` delete the `ServerAdminPassword=` line and let it regenerate.

does not resolve for me.
directly on startup the entry gets generated correctly at the last position of [ServerSettings] but as soon the server fully booted is puts again `ServerAdminPassword=#####?ServerPVE=True?ServerPassword=#####`

Log1x commented:
I can't test right now but try `-WinLiveMaxPlayers={{MAX_PLAYERS}}` for players.

Sergentval commented:
> I can't test right now but try `-WinLiveMaxPlayers={{MAX_PLAYERS}}` for players. From what I'm seeing, it should do the trick.
> 
> 
> 
> A lot of these are probably useless or don't do anything, but here's a dump of potential flags for reference (does not contain all possible flags):
> 
> 
> 
> ```
> 
> -SteamConnectIP=
> 
> -SteamServerName=
> 
> -audiomixer
> 
> -LevelSequence=
> 
> -MoviePipelineConfig=
> 
> -MoviePipelineLocalExecutorClass=
> 
> -MoviePipelineClass=
> 
> -MovieSceneCaptureType=
> 
> -MovieSceneCaptureManifest=
> 
> -ExecutorPythonClass=
> 
> -deterministicaudio
> 
> -TotalConversionMod=
> 
> -MapModID=
> 
> -mods=
> 
> -dev
> 
> -MyRedisServerId=
> 
> -NitradoQueryPort=
> 
> -WinLiveMaxPlayers=
> 
> -MovieQuality=
> 
> -VideoReplayCompressionImageFileFormat=
> 
> -server
> 
> -listenserver
> 
> -TermUser=
> 
> -TermState=
> 
> -foregroundworkers=
> 
> -physicalcorelimit=
> 
> -corelimit=
> 
> -iniFile=
> 
> -CustomConfig=
> 
> -Multiprocess
> 
> -backup-
> 
> -saveddirsuffix=
> 
> -enginesaveddirsuffix=
> 
> -Login=
> 
> -NoAnalytics 
> 
> -ImplicitSend
> 
> -Unattended
> 
> -AppName=
> 
> -CrashGUID=
> 
> -DebugSymbols=
> 
> -abslog=
> 
> -basedir=
> 
> -BaseFromWorkingDir
> 
> -abandoned
> 
> -cached
> 
> -extended
> 
> -noretire
> 
> -FileHostIP=
> 
> -ZenStoreHost=
> 
> -CookOnTheFlyServerWaitTime=
> 
> -forcerescan
> 
> -ignoreDenyLists
> 
> -asdir
> 
> -asfile
> 
> -log
> 
> -ARDiscoverThreads=
> 
> -ARDiscoverMinBatchSize=
> 
> -REPLAYSTREAMER=
> 
> -REPLAYSTREAMEROVERRIDE=
> 
> -CustomRenderPasses=
> 
> -CaptureGamut=
> 
> -HDRCompressionQuality=
> 
> -CaptureFramesInHDR=
> 
> -DisableScreenPercentage=
> 
> -PostProcessingMaterial=
> 
> -NOSCREENMESSAGES
> 
> -MovieFolder=
> 
> -MovieName=
> 
> -MovieOverwriteExisting=
> 
> -MovieRelativeFrames=
> 
> -HandleFrames=
> 
> -MovieEngineScalabilityMode=
> 
> -MovieCinematicMode=
> 
> -PathTracer=
> 
> -PathTracerSamplePerPixel=
> 
> -MovieFormat=
> 
> -ImageCaptureProtocol=
> 
> -AudioCaptureProtocol=
> 
> -MovieFrameRate=
> 
> -AudioCallbackBufferFrameSize=
> 
> -AudioNumBuffersToEnqueue=
> 
> -map=
> 
> -SystemSettings=
> 
> -display
> 
> -InjectionSandbox=
> 
> -Sandbox=
> 
> -InjectedSandbox=
> 
> -oldconsole 
> 
> -servergamelog 
> 
> -servergamelogincludetribelogs 
> 
> -NotifyAdminCommandsInChat
> 
> ```

Can be useful, they put a list to see which commands can be take by ARK SA right there: [Wiki Server](https://ark.wiki.gg/wiki/Server_configuration)

Sergentval commented:
![image](https://github.com/parkervcp/eggs/assets/40761828/09f59c30-91e0-4672-aeb6-27eb32e84e9e)

I guess it's 0 to disable ?

Can be cool to tell like the other one (auto updater)

Sergentval commented:
> ![image](https://github.com/parkervcp/eggs/assets/40761828/09f59c30-91e0-4672-aeb6-27eb32e84e9e)
> 
> I guess it's 0 to disable ?
> 
> Can be cool to tell like the other one (auto updater)

![image](https://github.com/parkervcp/eggs/assets/40761828/951effc8-24cd-458b-b861-a02d84b59123)

I'm not sure cause I can't launch the game for check but he say True in the starting line with the 0 I set in the first config of the server (fresh one)

dagbs commented:
> I'm not sure cause I can't launch the game for check but he say True in the starting line with the 0 I set in the first config of the server (fresh one)

I wrote the egg with the assumption those who are going to use it will be using a default ptero install without panel mods/addons. So, you're on your own.

gOOvER commented on game_eggs/steamcmd_servers/ark_survival_ascended/egg-ark--survival-ascended.json:
```diff
@@ -0,0 +1,154 @@
+    },
+    "scripts": {
+        "installation": {
+            "script": "#!\/bin\/bash\r\n# steamcmd Base Installation Script\r\n#\r\n# Server Files: \/mnt\/server\r\n# Image to install with is 'ubuntu:18.04'\r\napt -y update\r\napt -y --no-install-recommends --no-install-suggests install curl lib32gcc-s1 ca-certificates\r\n\r\n## just in case someone removed the defaults.\r\nif [ \"${STEAM_USER}\" == \"\" ]; then\r\n    STEAM_USER=anonymous\r\n ...
```
You are using an old steamcmd install script. pls update to:
https://github.com/parkervcp/eggs/blob/master/scripts/steamcmd_installer.sh

dagbs replied:
https://github.com/parkervcp/eggs/pull/2529/commits/37105e3885d81ebccc5a39c62d1be9ba1f69a400

vp-en commented:
I think there's something up with the `Server Name` variable. 

It seems that if you create a server with a `Server Name` that has spaces in it, the client won't be able to connect to it. I just tried creating a server named `Test Server 001`, and it wouldn't let me connect. But naming it `TestServer001` let me connect instantly. 

It might need to be parsed in `"` marks or something? I haven't tested it further, just noticed it

BrianWolvers commented:
> I think there's something up with the `Server Name` variable.
> 
> It seems that if you create a server with a `Server Name` that has spaces in it, the client won't be able to connect to it. I just tried creating a server named `Test Server 001`, and it wouldn't let me connect. But naming it `TestServer001` let me connect instantly.
> 
> It might need to be parsed in `"` marks or something? I haven't tested it further, just noticed it

It does say in the documentation on the ark wiki that " are needed for names with spaces.

joolas85 commented:
Hi, I have a problem that while the server is loading the ram usage and cpu performance drops and the server won't start

![kép](https://github.com/parkervcp/eggs/assets/110283368/0f009139-b0cd-4efb-b271-17ae476936ac)

gOOvER reviewed (changes requested):
Since its a Windows Game, you must add variable "WINDOWS_INSTALL"

joolas85 commented:
So far so good.
Where should I add it can you help me or is there a new egg that has it fixed?

joolas85 commented:
![kép](https://github.com/parkervcp/eggs/assets/110283368/4f7d163e-341d-4983-9b77-fc5644479888)

This is the Crash log

gOOvER commented:
> So far so good. Where should I add it can you help me or is there a new egg that has it fixed?

ehhhmm, thas a PR, so the latest is everytime here

vp-en commented:
> So far so good. Where should I add it can you help me or is there a new egg that has it fixed?

You can find the newest version of the PR'd egg here:
https://github.com/dagbs/eggs/blob/ark-survival-ascended/game_eggs/steamcmd_servers/ark_survival_ascended/egg-ark--survival-ascended.json

Sergentval commented:
I try this eggs for like 5 hours still a issue but can't find if this from the eggs or directly from the server file itself, like the server didn't save file if you shutdown you have some roll-back

parkervcp commented:
> I try this eggs for like 5 hours still a issue but can't find if this from the eggs or directly from the server file itself, like the server didn't save file if you shutdown you have some roll-back

I can almost guarantee that it's part of the shutdown and why the original ark egg uses the rcon tool for the cli.

Sergentval commented:
![image](https://github.com/parkervcp/eggs/assets/40761828/df14238b-f068-416e-97f4-addef7650696)
i can't access to the admin password i guess the problem can come from here ? anyone try the admin password have the same issue ?

dagbs commented:
![image](https://github.com/parkervcp/eggs/assets/133905860/6bdbef2d-3662-4dc4-a629-c9df107925c9)

and there you have it, https://github.com/parkervcp/eggs/pull/2529/commits/37105e3885d81ebccc5a39c62d1be9ba1f69a400
there is nothing else I can do to tune it out unless someone can figure out how to get mods working with a flag using variables, I'm done making changes.

Sergentval commented:
> ![image](https://user-images.githubusercontent.com/133905860/278882487-6bdbef2d-3662-4dc4-a629-c9df107925c9.png)
> 
> and there you have it, [37105e3](https://github.com/parkervcp/eggs/commit/37105e3885d81ebccc5a39c62d1be9ba1f69a400) there is nothing else I can do to tune it out unless someone can figure out how to get mods working with a flag using variables, I'm done making changes.

![image](https://github.com/parkervcp/eggs/assets/40761828/952385c0-45fa-4182-a004-b751bbbecf71)

Am I missing something? I've got the RCON on 37015.

IxPrumxI commented:
The admin password is messed up. Check the config

في أحد، ٢٩ أكتوبر، ٢٠٢٣ في ٨:٠٣ م، كتب Sergentval ***@***.***
>:

> [image: image]
> <https://user-images.githubusercontent.com/133905860/278882487-6bdbef2d-3662-4dc4-a629-c9df107925c9.png>
>
> and there you have it, 37105e3
> <https://github.com/parkervcp/eggs/commit/37105e3885d81ebccc5a39c62d1be9ba1f69a400>
> there is nothing else I can do to tune it out unless someone can figure out
> how to get mods working with a flag using variables, I'm done making
> changes.
>
> [image: image]
> <https://user-images.githubusercontent.com/40761828/278885423-952385c0-45fa-4182-a004-b751bbbecf71.png>
>
> Am I missing something? I've got the RCON on 37015.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/parkervcp/eggs/pull/2529#issuecomment-1784168295>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AII3NXSPT65VSMJZ5E6HDXLYBZ44RAVCNFSM6AAAAAA6UJQHTCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTOOBUGE3DQMRZGU>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

IxPrumxI commented on game_eggs/steamcmd_servers/ark_survival_ascended/egg-ark--survival-ascended.json:
```diff
@@ -0,0 +1,164 @@
+        "ghcr.io\/parkervcp\/steamcmd:proton": "ghcr.io\/parkervcp\/steamcmd:proton"
+    },
+    "file_denylist": [],
+    "startup": "rmv() { echo \"stopping server\"; rcon -t rcon -a 127.0.0.1:${RCON_PORT} -p ${ARK_ADMIN_PASSWORD} SaveWorld && rcon -t rcon -a 127.0.0.1:${RCON_PORT} -p ${ARK_ADMIN_PASSWORD} DoExit && wait ${ARK_PID}; echo \"Server Closed\"; exit; }; trap rmv 15 2; proton run .\/ShooterGame\/Binaries\/Win64\/ArkAscendedServer.exe \"{{SERVER_MAP}}?listen?MaxPlayers={{MAX_PLAYERS}}?SessionName={{SE ...
```
`?ServerAdminPassword=\"{{ARK_ADMIN_PASSWORD}}\"`
I just tried on a fresh install of the latest changes it it still grabs the ServerPassword 
![image](https://github.com/parkervcp/eggs/assets/34715358/c83c7d64-7dcd-4f5f-a229-84ff9145a2b4)

Sergentval replied:
Yeah i have the same lmao

dagbs replied:
fresh install != fresh re-install

Please make sure you test by deleting the entire server and re-creating it.

IxPrumxI replied:
Yes that's what I mean by a fresh install. I donwloaded the latest egg and imported it under a new nest to make sure I don't mix them up. Then I created a new server which had the problem.

dagbs replied:
https://github.com/parkervcp/eggs/pull/2529/commits/eacb24563c4b14c1389dbc5762ad8418d9e544f6
Please let me know if latest commit fixes the issues. Also, please make sure you update your existing server(s) with the latest Startup Command from the egg.

IxPrumxI commented:
![image](https://github.com/parkervcp/eggs/assets/34715358/a2c6112a-4e5c-4733-87c4-9470aa059fcd)
![image](https://github.com/parkervcp/eggs/assets/34715358/d4a03540-2a97-4039-8ac1-041f7857d8e3)

I was able to add the mods variable.
Here the egg with the mods and the fix for the ServerAdminPassword.
It still needs the description tho.
[egg-a-r-k--survival-ascended.json](https://github.com/parkervcp/eggs/files/13198760/egg-a-r-k--survival-ascended.json)

joolas85 commented:
Hello.
Where can I find fresh egg?

Sergentval commented:
![image](https://github.com/parkervcp/eggs/assets/40761828/635bcefb-4eca-4072-9a5b-1105f430b0b1)

I can't figure out where the RCON problem is coming from, is the 127.*.*.* for it to talk internally in the docker?

vp-en commented:
> ![image](https://user-images.githubusercontent.com/40761828/278899281-635bcefb-4eca-4072-9a5b-1105f430b0b1.png)
> 
> I can't figure out where the RCON problem is coming from, is the 127._._.* for it to talk internally in the docker?

I had the same error, until I updated to the [latest version of the egg](https://github.com/dagbs/eggs/blob/ark-survival-ascended/game_eggs/steamcmd_servers/ark_survival_ascended/egg-ark--survival-ascended.json).

dagbs commented:
> ![image](https://user-images.githubusercontent.com/40761828/278899281-635bcefb-4eca-4072-9a5b-1105f430b0b1.png)
> 
> I can't figure out where the RCON problem is coming from, is the 127._._.* for it to talk internally in the docker?

To those that are having `ServerAdminPassword=...?...` issues and/or `authentication failed` issues, please delete your server, download the latest commit, replace the existing Nest-egg with the one from the latest commit, and create the server fresh (not a reinstall, a full fresh install).

This is the last time anyone should ask about this issue and has been fixed for a while.

vp-en commented:
> > ![image](https://user-images.githubusercontent.com/40761828/278899281-635bcefb-4eca-4072-9a5b-1105f430b0b1.png)
> > 
> > I can't figure out where the RCON problem is coming from, is the 127._._.* for it to talk internally in the docker?
> 
> To those that are having `ServerAdminPassword=...?...` issues and/or `authentication failed` issues, please delete your server, download the latest commit, replace the existing Nest-egg with the one from the latest commit, and create the server fresh (not a reinstall, a full fresh install).
> 
> This is the last time anyone should ask about this issue and has been fixed for a while.

I did use the latest commit, updated the egg from the newest file. For me, it happens after I reboot the server, not on the first launch. On the first launch, it works as intended. But when I reboot, that's when it appends the other stuff.

Sergentval commented:
> > > ![image](https://user-images.githubusercontent.com/40761828/278899281-635bcefb-4eca-4072-9a5b-1105f430b0b1.png)
> > > I can't figure out where the RCON problem is coming from, is the 127._._.* for it to talk internally in the docker?
> > 
> > 
> > To those that are having `ServerAdminPassword=...?...` issues and/or `authentication failed` issues, please delete your server, download the latest commit, replace the existing Nest-egg with the one from the latest commit, and create the server fresh (not a reinstall, a full fresh install).
> > This is the last time anyone should ask about this issue and has been fixed for a while.
> 
> I did use the latest commit, updated the egg from the newest file. Also wiping the server completely every time the egg updates. For me, it happens after I reboot the server, not on the first launch. On the first launch, it works as intended. But when I reboot, that's when it appends the other stuff.

Same for me, except that it come directly at the first boot after installation. and when i look in the GameUser i have the line after admin password.

i fresh install the server delete all eggs and add the eggs with the newest comit in a fresh nest create. 

idk what is wrong.

dagbs commented:
Okay, I would like to firstly apologize to those who are having issues... but I finally identify the issue.

If your SESSION_NAME (Server Name) doesn't have a space, it will not work. This is an ARK bug and I'm currently testing a workaround to allow for space/no-space server names. However, adding a space will fix the issues.

Sergentval commented:
> Okay, I would like to firstly apologize to those who are having issues... but I finally identify the issue.
> 
> If your SESSION_NAME (Server Name) doesn't have a space, it will not work. This is an ARK bug and I'm currently testing a workaround to allow for space/no-space server names. However, adding a space will fix the issues.

So, first of all, it's cool, that you don't need to apologize lol, it's not your fault. to have found it in the first place is great, as I was stubbornly deleting the password from the start line, very funny LMAO.

For the spaces, put " in the place of each space?

Mythotical commented:
I'm having the same issue with RCON connection being refused. Not sure why. I'm even trying to use Arkon tool from my desktop and it can't connect. I assume that I have to have all the ports assigned to the server or does that even matter? I don't know if this is an issue with the server or my end but when trying to connect to my server I get connection timed out message.

dagbs commented:
Latest commit https://github.com/parkervcp/eggs/pull/2529/commits/eacb24563c4b14c1389dbc5762ad8418d9e544f6 addresses the issues regarding `ServerAdminPassword=...?ServerPVE=...?ServerPassword=...` and issues with rcon auth failing. You may have spaces or no-spaces in your server name.

Sergentval reviewed (approved):
After checking everything all work done

Log1x commented on game_eggs/steamcmd_servers/ark_survival_ascended/README.md:
```diff
@@ -0,0 +1,28 @@
+| Game            | 7777    |
+| Raw UDP         | 7778    |
+| Query           | 27015   |
+| RCON            | 27020   |
```
`37015` is the current egg default. I personally use 27020 but not sure if this should get updated to reflect what the egg does by default?

Sergentval replied:
Or add a new only for local ? The other is like if u want to access Rcon from outside right ?

Log1x replied:
> Or add a new only for local ? The other is like if u want to access Rcon from outside right ?

You can only have 1 RCON port. It's just optional to assign the RCON port to a public IP vs `127.0.0.1` for internal/console use.

QuintenQVD0 commented on game_eggs/steamcmd_servers/ark_survival_ascended/egg-ark--survival-ascended.json:
```diff
@@ -0,0 +1,174 @@
+        },
+        {
+            "name": "Auto-update server",
+            "description": "This is to enable auto-updating for servers.\r\n\r\nDefault is 1. Set to 0 to disable automatic updates",
```
This makes litte to no sence, as it will become that blue switch icon that just show on or off, they do not see the 1 or 0

QuintenQVD0 commented on game_eggs/steamcmd_servers/ark_survival_ascended/egg-ark--survival-ascended.json:
```diff
@@ -0,0 +1,174 @@
+        },
+        {
+            "name": "Battle Eye",
+            "description": "Enable BattleEye\r\n\r\n0 to disable\r\n1 to enable\r\n\r\ndefault=\"1\"",
```
same here

QuintenQVD0 commented on game_eggs/steamcmd_servers/ark_survival_ascended/egg-ark--survival-ascended.json:
```diff
@@ -0,0 +1,174 @@
+            "description": "ARK steam app id for auto updates. Leave blank to avoid auto update.",
+            "env_variable": "SRCDS_APPID",
+            "default_value": "2430930",
+            "user_viewable": true,
```
Users should not see this

QuintenQVD0 commented on game_eggs/steamcmd_servers/ark_survival_ascended/egg-ark--survival-ascended.json:
```diff
@@ -0,0 +1,174 @@
+            "default_value": "2430930",
+            "user_viewable": true,
+            "user_editable": false,
+            "rules": "nullable|numeric",
```
This should be `required|numeric|in:2430930`

QuintenQVD0 commented on game_eggs/steamcmd_servers/ark_survival_ascended/egg-ark--survival-ascended.json:
```diff
@@ -0,0 +1,174 @@
+        },
+        {
+            "name": "App ID",
+            "description": "ARK steam app id for auto updates. Leave blank to avoid auto update.",
```
This should say something like: app id as end users should not see this.

dagbs replied:
I'm not exactly sure on the wording, but I did update the Input Rules so the wording here I think is mute.

QuintenQVD0 commented on game_eggs/steamcmd_servers/ark_survival_ascended/egg-ark--survival-ascended.json:
```diff
@@ -0,0 +1,174 @@
+            "default_value": "",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "nullable|string",
```
please add at least some sort of max to this, like 512 or 1024

dagbs replied:
afaik, the max is 70. so, I'll use that. if it changes in the future, we can update the egg

QuintenQVD0 commented on game_eggs/steamcmd_servers/ark_survival_ascended/egg-ark--survival-ascended.json:
```diff
@@ -0,0 +1,174 @@
+            "default_value": "70",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|numeric",
```
There should be a limit between like 0 en max that ark supports

Sergentval replied:
![image](https://github.com/parkervcp/eggs/assets/40761828/08532694-601f-46aa-b0ce-71e18fba965c)

At Nitrado they have a maximum of 200 slots, but on the ARK wiki there's no clear mention of the player limit.

QuintenQVD0 replied:
Set it to 200

QuintenQVD0 commented on game_eggs/steamcmd_servers/ark_survival_ascended/egg-ark--survival-ascended.json:
```diff
@@ -0,0 +1,174 @@
+            "name": "Server Admin Password",
+            "description": "",
+            "env_variable": "ARK_ADMIN_PASSWORD",
+            "default_value": "changeme",
```
Default password values must be empty and forced by the admin to set with the required rule

dagbs replied:
> Default password values must be empty and forced by the admin to set with the required rule

I removed the default and added a description to explain it's used for RCON (remote/console) and in-game EnableCheats

QuintenQVD0 commented on game_eggs/steamcmd_servers/ark_survival_ascended/egg-ark--survival-ascended.json:
```diff
@@ -0,0 +1,174 @@
+            "env_variable": "QUERY_PORT",
+            "default_value": "27015",
+            "user_viewable": true,
+            "user_editable": true,
```
users can not change port, this should be vieuw only

IxPrumxI replied:
This can be changed. I have 2 servers on the same ip. 
![image](https://github.com/parkervcp/eggs/assets/34715358/fabd2ac5-90a5-4e7f-ae06-2e8faf1efb5e)
![image](https://github.com/parkervcp/eggs/assets/34715358/f29161d9-00ac-4067-851a-bd1f9d7cbdd8)

Sergentval replied:
Yeah it's important that the user can change the ports as needed.

QuintenQVD0 replied:
No , with pterodactyl that is not allowed. Ports must be set by the admin

gOOvER replied:
no its not. the User cant assign other Ports. only Admin can do this

IxPrumxI replied:
So if they got assigned a different port than the defaults they have to wait for an admin to change their server? They also might request an additional port and it might not be the same as default ports.

gOOvER replied:
the ports should be all empty and requiered. That would bei the best option

dagbs replied:
Admins can add/remove ports, as well as allocate additional ports. If end-users want to cycle through them... for whatever reason. I do not want to put restrictions on what they can change to but, regardless, they can't use a port that's not assigned to their server.

I did set min/max so it still requires a valid port but, other than this, it should be left up to the end-users.

QuintenQVD0 commented on game_eggs/steamcmd_servers/ark_survival_ascended/egg-ark--survival-ascended.json:
```diff
@@ -0,0 +1,174 @@
+            "default_value": "",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "nullable|string",
```
same here put at least some max here

Sergentval replied:
How do you determine a maximum that won't hinder the massive addition of mods, since mod ids are all different and can be more or less long no?

QuintenQVD0 commented:
If the requested changes are made then this can be merged.

IxPrumxI commented on game_eggs/steamcmd_servers/ark_survival_ascended/egg-ark--survival-ascended.json:
```diff
@@ -0,0 +1,174 @@
+            "env_variable": "RCON_PORT",
+            "default_value": "37015",
+            "user_viewable": true,
+            "user_editable": true,
```
This also can be changed and it works. Ark doesn't require specific ports to function.

Sergentval commented on game_eggs/steamcmd_servers/ark_survival_ascended/egg-ark--survival-ascended.json:
```diff
@@ -0,0 +1,174 @@
+        },
+        {
+            "name": "Server PvE",
+            "description": "Player-vs-Environment (PvE) or Player-vs-Player (PvP); default is (1) PvE enabled; disabling (0) will enable PvP",
```
I recommend this type of description to make things clearer too, what do you think ? 

**ON = Player-vs-Environment (PvE) OFF= Player-vs-Player-vs-Environment (PvPvE); By default is on.**

dagbs replied:
> I recommend this type of description to make things clearer too, what do you think ?
> 
> **ON = Player-vs-Environment (PvE) OFF= Player-vs-Player-vs-Environment (PvPvE); By default is on.**

I mean technically it's PvPvE and I prefer that but in-game it shows PvP, so I'm sticking with PvP in this case. I did update the format to what you posted since its easier to read.

QuintenQVD0 commented:
@dagbs I gave you a litte help, just please do a last test with this and fix the readme

dagbs commented on game_eggs/steamcmd_servers/README.md:
```diff
@@ -6,8 +6,9 @@ This is a collection of servers that use SteamCMD to install.
 [7 Days To Die](7_days_to_die)
 
-## ARK: Survival Evolved
+## ARK: Survival Ascended/Evolved
```
even though they are very VERY close to being the same game, its technically two entirely different games. so, I'll put them in their own game/folder

dagbs commented:
> @dagbs I gave you a litte help, just please do a last test with this and fix the readme

:\ I was making the changes and verifying your suggestions, doing a push, when I realized someone forced me to do a merge against my own branch.

dagbs commented:
> @dagbs I gave you a litte help, just please do a last test with this and fix the readme

I made all the changes to cleanup for a merge into master.

QuintenQVD0 commented:
> @dagbs I gave you a litte help, just please do a last test with this and fix the readme

You literly overwrotz all changes i made for you.

seba1114 commented:
After the latest update, the gameusersettings.ini and game.ini files return to default settings and I cannot change the server configuration in these files.

QuintenQVD0 reviewed (approved):
It looks good to me, it ran and started. I do not have the game to test but others did. So a big thank you for everyone who helped and the author of this PR ❤

gOOvER commented:
got the server working with wine stable and staging:

https://haste.goover.de/jolizenigo.yaml

gOOvER commented:
got it working with wine:

https://haste.goover.de/solupevuji.yaml

joolas85 commented:
![kép](https://github.com/parkervcp/eggs/assets/110283368/670d1752-1865-4ab8-8e7b-e0c5c44078c0)

If I delete Tribe data the server will start.
I have found that the master data has increased.
When I deleted this 1 trib the server was up and running again.

Any idea?

Log1x commented:
Known bug. Restore from a backup.

joolas85 commented:
We tried that too.
It's like the tribe log is damaged.
I was able to restore an older tribe log.

QuintenQVD0 commented:
Please use the discord for this and not closed pull request.
